csvcubed/csvcubed/schema/cube-config-schema.jsonc

```diff
@@ -0,0 +1,303 @@
+					"type": "string",
+                    "default": "dimension"
+                },
+                "label": {
+                    "description": "Replacement label for the Dimension",
+                    "type": "string"
+                },
+                "description": {
+                    "description": "Description for the column",
+                    "type": "string"
+                },
+                "from_existing": {
+                    "description": "URI of an Existing Qb Dimension",
+                    "type": "string",
+                    "format": "uri"
+                },
+
+                "definition_uri": {
+                    "description": "Source URI for the definition of the Dimension",
+                    "type": "string",
+                    "format": "uri"
+                },
+                "cell_uri_template": {
+                    "description": "Source URI for the code-list referred to by the dimension, e.g. 'http://example.org/code-list/{+some_column}'",
+                    "type": "string",
+                    "format": "uri"
+                }
+            }
+        },
+
+
+
+
+		"existingAttribute": {
+            "description": "An Existing Attribute Column",
+            "$ref": "#/$defs/attribute"
+		},
+
+		"newAttribute": {
+			"description": "A New Attribute column",
+			 "$ref": "#/$defs/attribute"
+		},
+
+		"attribute": {
+			"type": "object",
+			"required": ["type"],
+			"properties": {
+				"type": {
+                    "description": "The type of column",
+                    "type": "string",
+                    "default": "attribute"
+                },
+				"description": {
+					"description": "A description of the attribute",
+					"type": "string"
+				},
+
+                "from_existing": {
+                    "description": "The URI of an existing attribute column",
+                    "type": "string",
+                    "format": "uri"
+                },
+
+				"label": {
+                    "description": "The replacement label for the attribute",
+                    "type": "string"
+                },
+
+				"definition_uri":  {
+                    "description": "The URI that defines the attribute source",
+                    "type": "string",
+                    "format": "uri"
+                },
+				"required": {
+					"description": "optional, if set, the attribute requires values for each observation",
+					"type": "boolean"
+				},
+				"data_type": {
+					"description": "The data-type for the values for this attribute",
+					"type": "string"
+				},
+				"values": {
+					"description": "A list of the values for this attribute column",
+					"type": "array",
+					"items": {"$ref": "#/$defs/attributeValue"}
+				}
+			}
+        },
+
+		"attributeValue": {
+			"description": "A single attribute value, optionally referencing existing values and definition URIs",
+			"type": "object",
+			"properties": {
+				"label": {
```

Comment:
`label` should be required in attribute values/units/measures.

Reply:
> The contents of a column

When I define a list of units against a multi-units column, how do you know which unit definition matches up with which label?

When i define a unit against an observation column, where on earth does the unit's label come from?